src/jit/codegencommon.cpp

```diff
@@ -11109,45 +11109,7 @@ void                CodeGen::genRestoreCalleeSavedFltRegs(unsigned lclFrameSize)
 #if FEATURE_MULTIREG_STRUCTS
 CORINFO_CLASS_HANDLE Compiler::GetStructClassHandle(GenTreePtr tree)
 {
-    if (tree->TypeGet() == TYP_STRUCT)
-    {
-        // If the following assert fires (and we have a GT_COMMA), we'll probably need code like this:
-        //      while (tree->OperGet() == GT_COMMA) tree = tree->gtOp.gtOp2;
-
-        assert(tree->OperGet() != GT_COMMA);
-
-        switch (tree->OperGet())
-        {
-        default:
-            break;
-        case GT_CALL:
-            return tree->gtCall.gtRetClsHnd;
-
-        case GT_LDOBJ:
-            return tree->gtLdObj.gtClass;
-
-        case GT_LCL_VAR:
-        case GT_LCL_FLD:
-            return lvaTable[tree->gtLclVarCommon.gtLclNum].lvVerTypeInfo.GetClassHandle();
-
-        case GT_ARGPLACE:
-            return tree->gtArgPlace.gtArgPlaceClsHnd;
-
-        case GT_ASG:
-            assert(tree->gtOp.gtOp1->gtOper == GT_LCL_VAR || tree->gtOp.gtOp1->gtOper == GT_LCL_FLD);
-            return GetStructClassHandle(tree->gtOp.gtOp1);
-
-#if FEATURE_MULTIREG_STRUCT_RET
-        case GT_RET_EXPR:
-            return tree->gtRetExpr.gtRetClsHnd;
-
-        case GT_RETURN:
-            assert(tree->gtOp.gtOp1->gtOper == GT_LCL_VAR);
-            return GetStructClassHandle(tree->gtOp.gtOp1);
-#endif
-        }
-    }
-    return NO_CLASS_HANDLE;
+    return gtGetStructHandleIfPresent(tree);
```

Comment:
We can clean up this routine as there are only handful of references to it in codegencommon.cpp, compiler.h and morph.cpp.